### DISCUSSION TITLE
Backwards Compatibility Between CSI snapshots and the old In-Tree snapshots?

### DISCUSSION THREAD

**Discussion by arianitu** [vmware-tanzu/velero#3337] [Community support Q&A]
**Describe the problem/challenge you have**
After switching to CSI backups/restores and you have a backup made from the old in-tree backups, such as `velero-plugin-gcp`, will restores still work?

We want to know since the in-tree volume drivers will be removed in Kubernetes `1.21` and we need to know what happens with our old backups?

Thank you!

**Comment by nrb** [comment]
@arianitu This is feature we'll need to test for movement between 1.20 to 1.21.

As of today, with CSI snapshot support being beta, we do not have a translation layer that makes a VolumeSnapshotContents for each Velero-native snapshot. I think such a wrapper will be required to do this move.
